docs/elasticsearch/search-template.md

```diff
@@ -0,0 +1,416 @@
+        "play_name": "{% raw %}{{play_name}}{% endraw %}"
+      }
+    }
+  },
+  "params": {
+    "play_name": "Henry IV"
+  }
+}
+```
+
+
+## Save and execute search templates
+
+Once you have the search template working the way you want it to, you can save the source of that template as a script making it reusable for different input parameters.
+
+When saving the search template as a script, you need to specify the `lang` parameter to be `mustache`:
+
+```json
+POST _scripts/play_search_template
+{
+  "script": {
+    "lang": "mustache",
+    "source": {
+      "from": "{% raw %}{{page}}{{^page}}0{{/page}}{% endraw %}",
+      "size": "{% raw %}{{size}}{{^size}}10{{/size}}{% endraw %}",
+      "query": {
+        "match": {
+          "play_name": "{{play_name}}"
+        }
+      }
+    },
+    "params": {
+      "play_name": "Henry IV"
+    }
+  }
+}
+```
+
+Once the template is saved, you can refer to it by the template name using the `id` parameter.
+You can reuse this source template over and over again for different input values.
+
+```json
+GET _search/template
+{
+  "id": "play_search_template",
+  "params": {
+    "play_name": "Henry IV",
+    "page": 0,
+    "size": 1
+  }
+}
+```
+#### Sample output
+
+```json
+{
+  "took": 7,
+  "timed_out": false,
+  "_shards": {
+    "total": 6,
+    "successful": 6,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 3205,
+      "relation": "eq"
+    },
+    "max_score": 3.641852,
+    "hits": [
+      {
+        "_index": "shakespeare",
+        "_type": "_doc",
+        "_id": "4",
+        "_score": 3.641852,
+        "_source": {
+          "type": "line",
+          "line_id": 5,
+          "play_name": "Henry IV",
+          "speech_number": 1,
+          "line_number": "1.1.2",
+          "speaker": "KING HENRY IV",
+          "text_entry": "Find we a time for frighted peace to pant,"
+        }
+      }
+    ]
+  }
+}
+```
+
+If you have a stored template and want to validate it, you can use the `render` operation:
+
+```json
+POST _render/template
+{
+  "id": "play_search_template",
+  "params": {
+    "play_name": "Henry IV"
+  }
+}
+```
+
+#### Sample output
+
+```json
+{
+  "template_output": {
+    "from": "0",
+    "size": "10",
+    "query": {
+      "match": {
+        "play_name": "Henry IV"
+      }
+    }
+  }
+}
+```
+
+## Advanced parameter conversion with search templates
+
+You have a lot of different syntax options in Mustache to transpose the input parameters into a query.
+You can specify conditions, run loops, join arrays, convert arrays to JSON, and so on.
+
+### Conditions
+
+You can use the section tag in Mustache to represent conditions:
+
+```json
+{% raw %}{{#var}}var{{/var}}{% endraw %}
+```
+
+When `var` is a Boolean value, this syntax acts as an `if` condition. The `{% raw %}{{#var}}{% endraw %}` and `{% raw %}{{/var}}{% endraw %}` tags insert the values placed between them only if `var` evaluates to `true`.
```

Comment:
I strenuously object to the capitalization of boolean for the simple reason that it'd be ridiculous to capitalize Integers and Characters when discussing other primitive data types. I know the majority of style guides disagree with me on the grounds that the boolean was named after a person, but they're indefensibly wrong. By their logic, we should capitalize shrapnel and saxophone, too.